src/pages/ProductDetailPage/ProductDetail.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import './ProductDetail.css';
+import { useParams , useNavigate } from 'react-router-dom';
+import { useProducts } from '../../context/Product-context';
+import { useToastContext } from '../../context/toastContext';
+import { useCartlist } from '../../context/cart-context';
```

Comment:
suggestion: create a index.js file in context folder and export all contexts form it.